### ISSUE TITLE
Feature Request: Signal `tockloader listen` processes

### ISSUE THREAD

**Issue by bradjc** [helena-project/tockloader#10]
In my experience any tockloader command will fail if another terminal has `tockloader listen` running. Therefore it would be nice if on any command, tockloader would signal all other tockloader processes, and if those were `tockloader listen` on the same port, they would release the serial port, wait for the signalling process to finish, and then resume listen. This would make tockloader work seamlessly with itself.

Note: I'm not actually sure what the best way to get tockloader processes to communicate is.

**Comment by alevy** [comment]
pid file? somewhere in the user's home directory? perhaps named by the serial port as well so it only interrupts processes referring to the same serial port?

**Comment by ppannuto** [comment]
Okay, so I spent entirely too long today trying to get this to work (headway in `listen-ipc` branch), and ended up failing.

The biggest reason I couldn't get this to work is that miniterm isn't designed to be killed by another thread. It's reads are blocking until there's some form of I/O and there's no way to cancel them prematurely without somehow injecting characters into stdin.

The next route I tried to go down was just killing off the process entirely, first by using `execvp`, but that meant that miniterm's atexit handlers didn't run and your terminal was corrupt. Next I tried some hackery around `spawnvp`ing a fresh copy and completely killing off the original process, but that didn't work either.

Despite sounding simple, this feature is proving non-trivial
